arangod/Replication/DatabaseInitialSyncer.cpp

```diff
@@ -210,130 +209,138 @@ arangodb::Result fetchRevisions(
   std::size_t current = 0;
   auto guard = arangodb::scopeGuard(
       [&current, &stats]() noexcept { stats.numDocsRequested += current; });
+
   char ridBuffer[arangodb::basics::maxUInt64StringSize];
-  std::unique_ptr<arangodb::httpclient::SimpleHttpResult> response;
-  while (current < toFetch.size()) {
-    arangodb::transaction::BuilderLeaser requestBuilder(&trx);
-    {
-      VPackArrayBuilder list(requestBuilder.get());
-      for (std::size_t i = 0; i < 5000 && current + i < toFetch.size(); ++i) {
-        requestBuilder->add(toFetch[current + i].toValuePair(ridBuffer));
+  std::deque<arangodb::network::FutureRes> futures;
+
+  arangodb::network::ConnectionPool* pool = netFeature.pool();
+
+  while (current < toFetch.size() || !futures.empty()) {
+    // Send some requests off if not enough in flight and something to go
+    while (futures.size() < 10 && current < toFetch.size()) {
+      VPackBuilder requestBuilder;
+      {
+        VPackArrayBuilder list(&requestBuilder);
+        std::size_t i;
+        for (i = 0; i < 5000 && current + i < toFetch.size(); ++i) {
+          requestBuilder.add(toFetch[current + i].toValuePair(ridBuffer));
+        }
+        current += i;
       }
-    }
-    std::string request = requestBuilder->slice().toJson();
-
-    double t = TRI_microtime();
-    config.connection.lease(
-        [&](arangodb::httpclient::SimpleHttpClient* client) {
-          response.reset(client->retryRequest(arangodb::rest::RequestType::PUT,
-                                              url, request.data(),
-                                              request.size(), headers));
-        });
-    stats.waitedForDocs += TRI_microtime() - t;
```

Comment:
`stats.waitedForDocs` will now always be 0. Can we populate it somehow, to get some insights into how long the document fetching takes?

Reply:
Good point, I will think of something. Maybe only take the actual waiting time for futures.

Reply:
Done.